### PR TITLE
Kyle

### PR DESCRIPTION
**Pull request by kbricker8**
Added sign in page as well as log out functionality.

### CODE REVIEW

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java [hunk 1]
```diff
@@ -25,6 +25,12 @@ protected void onCreate(Bundle savedInstanceState) {
 
         FirebaseFirestore db;
 
+        if (!((GroupProject) this.getApplication()).isSignedIn()) {
+            Intent intent = new Intent(this, SignInActivity.class);
+            ((GroupProject) this.getApplication()).setSignedIn(true);
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java:30]: Set signedIn to true after actually signing in.

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java [hunk 2]
```diff
@@ -102,6 +106,11 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
             db.collection("Users").document(thisUser.getUsername())
                     .set(user, SetOptions.merge());
         });
+
+        signOutbtn.setOnClickListener(view1 -> {
+            Intent intent = new Intent(thisContext, SignInActivity.class);
+            startActivity(intent);
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java:112]: Set signIn to false here.

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/SignInActivity.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            ((GroupProject) this.getApplication()).setUsername(username);
+
+            if (!TextUtils.isEmpty(passwordEdit.getText().toString())) {
+                String password = passwordEdit.getText().toString();
+
+                Map<String, Object> user = new HashMap<>();
+                user.put("password", password);
+
+                db.collection("Users").document(username)
+                        .set(user, SetOptions.merge());
```
Comment (schen32) [inline CMPUT301Project/app/src/main/java/com/example/GroupProject/SignInActivity.java:49]: Here we are signing in while also updating the password. We should confirm the password and set signedIn to true here instead. Also, no register page yet?

File: CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java (hunk 1 above)
Comment (kbricker8) [reply CMPUT301Project/app/src/main/java/com/example/GroupProject/MainActivity.java:30]: The implementation of signedIn currently only affects anything on initial app start up. I'll change it incase the functionality is changed in the future though :)

**Pull request merged**